File: ObjC-SampleApp/SideEngineSample/SideEngineSample/VideoAskViewController.m
//
//  VideoAskViewController.m
//  SideEngineSample
//
//  Created by Phoenix@TNM#Mac on 07/07/23.
//

#import "VideoAskViewController.h"

@interface VideoAskViewController ()

@end

@implementation VideoAskViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // If video survey enable in the your partner
    if ([NSURL URLWithString:[BBSideEngineManager shared].surveyVideoURL]) {
        NSURLRequest *myRequest = [NSURLRequest requestWithURL:[NSURL URLWithString:[BBSideEngineManager shared].surveyVideoURL]];
        [self.webview loadRequest:myRequest];
    }
}

- (IBAction)closeTapped {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"" message:@"Are you sure you want to end this survey?" preferredStyle:UIAlertControllerStyleAlert];

    [alert addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:nil]];
    [alert addAction:[UIAlertAction actionWithTitle:@"End" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        if (self.delegate != nil) {
            [self.delegate didFinishSurvey];
        }
        dispatch_async(dispatch_get_main_queue(), ^{
            [self dismissViewControllerAnimated:YES completion:nil];
        });
    }]];

    dispatch_async(dispatch_get_main_queue(), ^{
        [self presentViewController:alert animated:YES completion:nil];
    });
}

@end

File: Swift-SampleApp/FlareSDKSample/SideEngineSample/VideoAskViewController.swift
//
//  VideoAskViewController.swift
//  SideEngineSample
//
//  Created by Phoenix Innovate on 30/11/22.
//

import UIKit
import BBSideEngine
import WebKit

protocol VideoAskDelegate {
    func didFinishSurvey()
}

class VideoAskViewController: UIViewController {
    var delegate: VideoAskDelegate?
    @IBOutlet weak var webview: WKWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        //If video survey enable in the your partner
        if let videoUrl = URL(string:BBSideEngineManager.shared.surveyVideoURL) {
            let myRequest = URLRequest(url: videoUrl)
            self.webview.load(myRequest)
        }
        
        // Do any additional setup after loading the view.
    }
    
    @IBAction func closeTapped() {
        let alert = UIAlertController(title: "", message: "Are you sure you want to end this survey?",         preferredStyle: .alert)
        
        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { _ in
            
        }))
        alert.addAction(UIAlertAction(title: "End", style: .default, handler: { _ in
            if self.delegate != nil {
                self.delegate?.didFinishSurvey()
            }
            DispatchQueue.main.async {
                self.dismiss(animated: true)
            }
        }))
        
        DispatchQueue.main.async {
            self.present(alert, animated: true, completion: nil)
        }
        
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
